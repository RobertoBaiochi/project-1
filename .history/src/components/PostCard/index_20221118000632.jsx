

const PostCard = ({title, cover, body, id}) => {
  
  return (
    <div className="post" key={post.id} >
      <img src={post.cover} alt={post.title} />
      <div className="post-content">
        <h1>{post.title}</h1>
        <p>{post.body}</p>
      </div>
    </div>
  )
}

export default PostCard