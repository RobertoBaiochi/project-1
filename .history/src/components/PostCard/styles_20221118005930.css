.post {
  background: #fff;
  box-shadow: 0 0 10px 2px rgba(0, 0, 0, 0.2);
  transition: transform 100ms ease-in-out;
  filter: brightness(0.90);
}

.post:hover {
  transform: scale(1.02);
  filter: brightness(1);
}

.post-content {
  padding: 1.75rem
}