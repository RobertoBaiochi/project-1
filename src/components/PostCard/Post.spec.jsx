import { render, screen } from "@testing-library/react";
import PostCard from ".";
import { postCardPropsMock } from "./mock";

import TestRenderer from 'react-test-renderer'

const props = postCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/title 1/i))
      .toHaveAttribute("src","img/img.png");
    
    expect(screen.getByRole("heading", { name: 'title 1 1' }))
      .toBeInTheDocument();

    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const testRenderer = TestRenderer.create(
        <PostCard {...props} ></PostCard>
      ).toJSON();
      
    expect(testRenderer).toMatchSnapshot();
});

});
