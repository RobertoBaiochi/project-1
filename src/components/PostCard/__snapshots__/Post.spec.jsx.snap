// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  className="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    className="post-content"
  >
    <h2>
      title 1
       
      1
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
